fw/http_match.c

```diff
@@ -231,18 +231,16 @@ match_host_forwarded(const TfwHttpReq *req, const TfwHttpMatchRule *rule)
 static bool
 match_host(const TfwHttpReq *req, const TfwHttpMatchRule *rule)
 {
-	const TfwStr *host = &req->host;
-
 	/*
 	 * TODO #1630: replace with a single condition
 	 * on the authority special header.
 	 */
-	if (host->len > 0) {
+	if (req->uri_host && req->uri_host->len > 0) {
 		/*
 		 * RFC 7230 5.4: Host header must
 		 * be ignored when URI is absolute.
 		 */
-		return host_val_eq(host, rule);
+		return host_val_eq(req->uri_host, rule);
```

Comment:
This is the only place where `uri_host` is used - why do we need it here? Why can't we just use `req->host` here?

According to the TODO comment above, this is the main point for #1630. I'd say this is exactly the place where `req->host`, picked from all URI auth, Host and :authority, should be used. Most likely some tests should be fixed along with the documentation https://github.com/tempesta-tech/tempesta/issues/1630#issuecomment-1500325853 .

We also need (and this would be good to cover with tests if we don't have such tests yet) to be able to get different matching results for `host == "foo" -> ...`, `hdr Host == "foo" ->` and `hdr :authority == "foo" -> ...` . Please test the case from https://github.com/tempesta-tech/tempesta/issues/1667 - the issue also should be fixed with this PR.